src/openrct2/ride/Vehicle.cpp

```diff
@@ -1525,13 +1525,13 @@ bool Vehicle::OpenRestraints()
         {
             if (vehicle->animationState <= 0xCCCC)
             {
-                vehicle->animationState += 0x3333;
+                vehicle->animationState += vehicleEntry->AnimationSpeed;
             }
             else
             {
                 vehicle->animationState = 0;
                 vehicle->animation_frame++;
-                vehicle->animation_frame &= 7;
+                vehicle->animation_frame %= vehicleEntry->AnimationFrames;
```

Comment:
This will now wrap instead of a cap, isn't this changing the logic?

Reply:
`&= 7` is the same as `%= 8`. Assuming that AnimationFrames is 8 here.

Reply:
Right I forgot power of two thingy since it was 7. Should be alright if this holds 8 i guess.

Reply:
the default value is 8 yes